packages/shared/src/components/squads/utils.ts

```diff
@@ -0,0 +1,42 @@
+import request from 'graphql-request';
+import { SourceData, SOURCE_QUERY } from '../../graphql/sources';
+import { apiUrl } from '../../lib/config';
+import classed from '../../lib/classed';
+import { Squad } from '../../graphql/squads';
+
+export async function checkSourceExists(id: string): Promise<boolean> {
+  try {
+    const data = await request<SourceData>(`${apiUrl}/graphql`, SOURCE_QUERY, {
+      id,
+    });
+    return !!data.source;
+  } catch (err) {
+    if (err?.response?.errors?.[0].extensions?.code === 'NOT_FOUND') {
```

Comment:
@idoshamun 

This is the error I get when we check for a source with no image:
```
Error: Cannot return null for non-nullable field Source.image.: 
```
```json

{
    "request": {
        "query": "\n  query Source($id: ID!) {\n    source(id: $id) {\n      id\n      image\n      name\n    }\n  }\n",
        "variables": {
            "id": "squad"
        }
    },
    "response": {
        "data": {
            "source": null
        },
        "status": 200,
        "errors": [
            {
                "message": "Cannot return null for non-nullable field Source.image.",
                "locations": [
                    {
                        "line": 5,
                        "column": 7
                    }
                ],
                "path": [
                    "source",
                    "image"
                ]
            }
        ],
        "headers": {
            "map": {
                "content-length": "167",
                "origin-agent-cluster": "?1",
                "content-security-policy": "default-src 'self';base-uri 'self';font-src 'self' https: data:;form-action 'self';frame-ancestors 'self';img-src 'self' data:;object-src 'none';script-src 'self';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests",
                "strict-transport-security": "max-age=15552000; includeSubDomains",
                "x-content-type-options": "nosniff",
                "cross-origin-embedder-policy": "require-corp",
                "date": "Fri, 30 Dec 2022 11:44:30 GMT",
                "referrer-policy": "no-referrer",
                "x-permitted-cross-domain-policies": "none",
                "content-type": "application/json; charset=utf-8",
                "cross-origin-opener-policy": "same-origin",
                "x-dns-prefetch-control": "off",
                "access-control-allow-origin": "http://webapp.local.com:5002",
                "x-download-options": "noopen",
                "vary": "Origin, Accept-Encoding",
                "access-control-allow-credentials": "true",
                "x-frame-options": "SAMEORIGIN",
                "connection": "close",
                "x-xss-protection": "0",
                "cross-origin-resource-policy": "same-origin"
            }
        }
    }
}
```

Reply:
@el it's not possible to have a source without an image. I need to fix the db constraint

Reply:
Ok, I am not going to do for this then 👍